src/utils/experimentUpdatesHandler.js

```diff
@@ -68,12 +70,18 @@ const onGEM2SUpdate = (update, dispatch, experimentId) => {
 };
 
 const onWorkResponseUpdate = (update, dispatch, experimentId) => {
-  const { request, response: { error } } = update;
+  const { request: { body }, response: { error } } = update;
 
-  if (request.body.name === 'ClusterCells') {
-    if (error) throw new Error(error);
+  if (error) throw new Error(error);
```

Comment:
what was happening in the past when we weren't throwing an error for all types of body.name?

Reply:
Previously there was only 1 work request which uses this path, `ClusterCells`, so the error is handled inside an if statement which checks if the body of the work result is `ClusterCells` before checking the error. Before that, we did not use the `ExperimentUpdates-` channel for broadcasting completion of work request results.

Reply:
In this implementation, all types of messages returned from the API will be able to trigger an error. If we handle the error after we check the body name, we'd be able to check if there returned message is correct (has a valid body name) before acting on the error.